azure-iot-device/azure/iot/device/iothub/pipeline/mqtt_pipeline.py

```diff
@@ -389,13 +475,30 @@ def disable_feature(self, feature_name, callback):
         :param feature_name: one of the feature name constants from constant.py
 
         :raises: ValueError if feature_name is invalid
+        :raises: :class:`azure.iot.device.iothub.pipeline.exceptions.PipelineNotRunning` if the
+            pipeline has previously been shut down
+
+        The following exceptions are not "raised", but rather returned via the "error" parameter
+        when invoking "callback":
+
+        :raises: :class:`azure.iot.device.iothub.pipeline.exceptions.ConnectionFailedError`
+        :raises: :class:`azure.iot.device.iothub.pipeline.exceptions.ConnectionDroppedError`
+        :raises: :class:`azure.iot.device.iothub.pipeline.exceptions.UnauthorizedError`
+        :raises: :class:`azure.iot.device.iothub.pipeline.exceptions.ProtocolClientError`
         """
+        self._ensure_running()
         logger.debug("disable_feature {} called".format(feature_name))
         if feature_name not in self.feature_enabled:
             raise ValueError("Invalid feature_name")
-        self.feature_enabled[feature_name] = False
+        # TODO: What about if the feature is already disabled?
 
         def on_complete(op, error):
+            if error:
+                logger.error(
+                    "Unsubscribe for {} failed. Not disabling feature".format(feature_name)
```

Comment:
>Not disabling feature [](start = 48, length = 21)

I don't like this assumption.  If you fail to disable something, it doesn't (necessarily) mean that it's still enabled.  If lower code disables the feature and then raises an exception, the `feature_enabled` flag will be incorrect.  Because the code in sync_clients.py only calls `enable_feature()` if enabled==False, there's no way to recover from this error.

Reply:
Good point. I'll reorganize, and issue a warning instead